sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -1312,9 +1312,12 @@ RTDeviceBinaryImage *getBinImageFromMultiMap(
         reinterpret_cast<const char *>(&CompileTargetByteArray[0]),
         CompileTargetByteArray.size());
     // Note: there are no explicit targets for CPUs, so on x86_64,
-    // so we use a spir64_x86_64 compile target image.
+    // intel_cpu_spr, and intel_cpu_gnr, we use a spir64_x86_64
+    // compile target image.
     if ((ArchName == CompileTarget) ||
-        (ArchName == "x86_64" && CompileTarget == "spir64_x86_64")) {
+        ((ArchName == "x86_64" || ArchName == "intel_cpu_spr" ||
+          ArchName == "intel_cpu_gnr") &&
+         CompileTarget == "spir64_x86_64")) {
```

Comment:
This is not safe, we easily can forget to update this `if` with more CPU architectures. 
This API should be used here:
```
bool ext_oneapi_architecture_is(                       (2)
    ext::oneapi::experimental::arch_category category);
```
https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_oneapi_device_architecture.asciidoc#new-member-function-of-device-class

Reply:
Ahh, we don't have `intel_cpu` category yet. 
@gmlueck, is it ok to update the `device_architecture` spec and impl by extending `arch_category` with `intel_cpu`?

Reply:
Do we want a generic test for "Intel CPU" here?  For example, if the device image is compiled for "intel_cpu_spr", we can only use it if the CPU really is Sapphire Rapids.  Therefore, it seems like we might want a series of "if" statements here.  Is there some other place in the code where we check more precisely whether the ArchName matches the CPU model?  Should that logic be pushed down into the unified runtime?

Reply:
> This is not safe, we easily can forget to update this if with more CPU architectures.

I would say this special handling in this if statement is mainly temporary, as we currently have no dedicated targets for cpu targets in the compiler, i.e. there is no `-fsycl-targets=intel_cpu_spr` and `-fsycl-targets=intel_cpu_gnr` (nor `-fsycl-targets=x86_64`).

> For example, if the device image is compiled for "intel_cpu_spr", we can only use it if the CPU really is Sapphire Rapids

Once again, this is what we want, however, as far as I am aware, there is no way to compile for specifically for `intel_cpu_spr`; the generic `spir64_x86_64` target is used, and there is no `-fsycl-targets=intel_cpu_spr`.

Reply:
@dm-vodopyanov Added a TODO comment [7e5fa32](https://github.com/intel/llvm/pull/15208/commits/7e5fa32a66b71554829611f68c8033bacbe39f8b)